### PR TITLE
Add option to pass kubeconfig path to Flux 2 installer

### PR DESCRIPTION
**Pull request by Callisto13**
### Description

Somehow forgot to do this originally 🤦‍♀️ .

- New optional config field on flux object to specify non-default kubeconf path
- When installing flux as part of create where the config path has been changed, the flux installer config is updated to match
- Updated build dockerfile to install latest stable version of `kubectl`. Flux requires >= 1.18.0
- A couple of other things needed by flux integration tests

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: build/docker/Dockerfile [hunk 1]
```diff
@@ -32,8 +32,7 @@ RUN apk add --no-cache --initdb --root /out \
     openssh \
     && true
 
-ENV KUBECTL_VERSION v1.15.11
-RUN curl --silent --location "https://dl.k8s.io/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" --output /out/usr/local/bin/kubectl \
+RUN curl --silent --location "https://dl.k8s.io/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" --output /out/usr/local/bin/kubectl \
```
Comment (aclevername) [inline build/docker/Dockerfile:35]: nice

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 2]
```diff
@@ -867,6 +867,10 @@ type Flux struct {
 	// +optional
 	Namespace string `json:"namespace,omitempty"`
 
+	// Path to the kubernetes config for the cluster
+	// +optional
+	Kubeconfig string `json:"kubeconfig,omitempty"`
+
```
Comment (aclevername) [inline pkg/apis/eksctl.io/v1alpha5/types.go:873]: worth noting the behaviour when not set?

File: pkg/flux/flux.go [hunk 3]
```diff
@@ -58,8 +71,10 @@ func (c *Client) runFluxCmd(args ...string) error {
 
 func setEnvVars(tokenPath, gitProvider string) executor.EnvVars {
 	envVars := executor.EnvVars{
-		"PATH": os.Getenv("PATH"),
-		"HOME": os.Getenv("HOME"),
+		"PATH":                  os.Getenv("PATH"),
+		"HOME":                  os.Getenv("HOME"),
+		"AWS_ACCESS_KEY_ID":     os.Getenv("AWS_ACCESS_KEY_ID"),
+		"AWS_SECRET_ACCESS_KEY": os.Getenv("AWS_SECRET_ACCESS_KEY"),
```
Comment (aclevername) [inline pkg/flux/flux.go:77]: feel like I'm being dum and could just goggle it but why does flux need AWS creds?

**Review by aclevername — APPROVED** [review]
LGTM, 1 nit & 1 question 😄

File: pkg/apis/eksctl.io/v1alpha5/types.go (hunk 2 above)
Comment (Callisto13) [reply pkg/apis/eksctl.io/v1alpha5/types.go:873]: maybe? the behaviour isn't ours, in our case we just don't do anything, but i can note flux's default

File: pkg/flux/flux.go (hunk 3 above)
Comment (Callisto13) [reply pkg/flux/flux.go:77]: flux needs to auth against the cluster to run kubectl things

Comment (aclevername) [reply pkg/flux/flux.go:77]: Thank!

**Pull request merged**